File: translation_output/misleading_test/magicoder/avatar/java/python-sanitized/codeforces_514_A.py
def solve(x):
    list = []
    while x > 0:
        r = x % 10
        if 9 - r < r:
            if x // 10 == 0 and 9 - r == 0:
                list.append(r)
            else:
                list.append(9 - r)
        else:
            list.append(r)
        x = x // 10

    pow = 0
    newNumber = 0

    for i in range(len(list)):
        newNumber = newNumber + list[i] * (10 ** pow)
        pow += 1

    return newNumber

# Test input
x = 5728
print(solve(x))  # Expected output: 4304

File: Results/DER/Translation/translation_output/w_test/gpt-4/avatar/python/java/codeforces_514_A.java
```java
import java.util.Scanner;

public class PythonToJava {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String ip = sc.next();
        String st = "";

        if (ip.length() != 1) {
            if (ip.charAt(0) == '9') {
                st = "9";
                ip = ip.substring(1);
            }
            for (int i = 0; i < ip.length(); i++) {
                int digit = Character.getNumericValue(ip.charAt(i));
                if (digit > 4) {
                    int n = 9 - digit;
                    st += Integer.toString(n);
                } else {
                    st += ip.charAt(i);
                }
            }
        } else {
            st = ip;
        }
        System.out.println(Integer.parseInt(st));
    }
}
```

To run a test case for the input "5728" using Java, you can either:
1. Compile the above Java code and run it, entering the value when prompted, or 
2. Modify the code slightly to directly assign the test input as a string to the `ip` variable rather than reading from `System.in`. Here's how you might modify it for a quick test without user input:

```java
public class PythonToJava {
    public static void main(String[] args) {

        // Directly provide test input without using Scanner
        String ip = "5728"; // Test input
        String st = "";

        // Rest of the code remains unchanged...
        // ... Logic to generate output ...
        
        System.out.println(Integer.parseInt(st)); // Expected output: 4221
    }
}
```
